### PR TITLE
Add Encryption workflow for storage cluster creation

### PR DESCRIPTION
**Pull request by a2batic**
Screenshot from 2020-11-02 01-41-40

JIRA: https://issues.redhat.com/browse/RHSTOR-1445
Signed-off-by: Kanika Murarka <kmurarka@redhat.com>

### CODE REVIEW

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/kms-config/kms-config.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        <TextInput
+          value={serviceName}
+          onChange={getServiceName}
+          isRequired
+          type="text"
+          id="kms-service-name"
+          name="kms-service-name"
+        />
+      </FormGroup>
+    </>
```
[the remark this answers is not in the document]
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/kms-config.tsx:55]: There will be more `FormGroup` to be added https://github.com/openshift/console/pull/7062/files#diff-ae3d56459b2491d83b6fe9502d3c74b3910fc605496fef5d596dcf713a097af6R12

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx [hunk 2]
```diff
@@ -1,231 +0,0 @@
...
-      </Form>
-    </>
-  );
-});
-
-type CreateOCSProps = {
-  match: RouterMatch<{ appName: string; ns: string }>;
-  setIsNewSCToBeCreated?: React.Dispatch<boolean>;
-  setHasNoProvSC?: React.Dispatch<boolean>;
-};
```
[the remark this answers is not in the document]
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx:231]: ack

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 3]
```diff
@@ -1,23 +1,155 @@
...
+              className="ocs-install-encryption__form-checkbox"
+            />
+          </FormGroup>
+          {!nextValid && (
+            <Alert
+              variant="danger"
+              className="co-alert ocs-install-encryption__form-alert"
+              title="Select at least 1 encryption level or disable encryption"
+              isInline
+            />
```
[the remark this answers is not in the document]
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:129]: ack

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 4]
```diff
@@ -1,23 +1,155 @@
...
+          >
+            <Checkbox
+              id="cluster-wide-encryption"
+              isChecked={clusterWideChecked}
+              label={
+                <span className="ocs-install-encryption__pv-title--padding">
+                  Cluster-wide encryption
+                </span>
+              }
+              aria-label="Checkbox with Cluster-wide encryption"
```
[the remark this answers is not in the document]
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:108]: ack

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 5]
```diff
@@ -1,23 +1,155 @@
...
+              description="Encryption for the entire cluster (block and file)"
+              onChange={toggleClusterWideEncryption}
+              className="ocs-install-encryption__form-checkbox"
+            />
+            <Checkbox
+              id="pv-encryption"
+              isChecked={pvChecked}
+              label={<PVEncryptionLabel />}
+              aria-label="Checkbox with Cluster-wide encryption"
```
[the remark this answers is not in the document]
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:117]: ack

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/internal-mode/reducer.ts [hunk 6]
```diff
@@ -1,35 +1,59 @@
...
-  enableEncryption: boolean;
   enableMinimal: boolean;
+
+  // Encryption state declare
+  enableClusterWideEncryption: boolean;
+  enablePvEncryption: boolean;
+  hasEncryptionHandled: boolean;
+  kmsServiceName: string;
+
+  modeType: string;
```
[the remark this answers is not in the document]
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/internal-mode/reducer.ts:16]: ack

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 7]
```diff
@@ -1,23 +1,155 @@
...
+                  Cluster-wide encryption
+                </span>
+              }
+              aria-label="Checkbox with Cluster-wide encryption"
+              description="Encryption for the entire cluster (block and file)"
+              onChange={toggleClusterWideEncryption}
+              className="ocs-install-encryption__form-checkbox"
+            />
+            <Checkbox
+              id="pv-encryption"
```
[the remark this answers is not in the document]
Comment (a2batic) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:114]: ya, internally its encryption PVs, so I kept PV encryption. Will change it to storage class encyption for better understanding.

**Comment by a2batic** [comment]
@cloudbehl ack, will add these changes in KMS related PR https://github.com/openshift/console/pull/7153

**Comment by afreen23** [comment]
/lgtm

okay to take https://github.com/openshift/console/pull/7062#discussion_r519565433
in the other PR

**Pull request merged**